src/fileActions.jsx

```diff
@@ -134,7 +133,11 @@ const CreateDirectoryModal = ({ currentPath }) => {
     const [name, setName] = useState("");
     const [nameError, setNameError] = useState(null);
     const [errorMessage, setErrorMessage] = useState(undefined);
-    const createDirectory = () => spawnCreateDirectory(name, currentPath, Dialogs, setErrorMessage);
+    const createDirectory = () => {
+        const path = currentPath + name;
+        cockpit.spawn(["mkdir", path], { superuser: "try", err: "message" })
```

Comment:
This is moved code, but I find the unconditional `superuser: try` here problematic. Won't that create root-owned directories even if you run it in e.g. your home dir? I feel like this would better work the other way round: first try as your user, and if it fails on permission-denied, retry with root?

This of course becomes even more interesting if it's *someone else*'s home directory. Even a system user, like /var/lib/postgresql/. Perhaps the default should be to run it as the user who owns the containing directory?

Reply:
Filed as issue #390

Reply:
Yes, I am afraid this is the case. And this affects all items here, like rename etc.